flink/src/main/java/org/apache/iceberg/flink/FlinkSchemaUtil.java

```diff
@@ -21,10 +21,19 @@
 
 import org.apache.flink.table.api.TableSchema;
 import org.apache.flink.table.types.FieldsDataType;
+import org.apache.flink.table.types.logical.LogicalType;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.flink.table.types.utils.TypeConversions;
 import org.apache.iceberg.Schema;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.TypeUtil;
 
+/**
+ * Converter between Flink types and Iceberg type.
+ * The conversion is not a 1:1 mapping that not allows back-and-forth conversion. So some information might get lost
+ * during the back-and-forth conversion.
```

Comment:
Can you be more specific about this? What is a case where information is lost?

Reply:
If I understand correctly, this is lossy because Iceberg doesn't represent some types that Flink supports, like `CHAR(N)`. Is that right?

Reply:
Yes

Reply:
Iceberg to Flink: will loss UUID.
Flink to Iceberg: will loss precisions.